### PR TITLE
Signature server for proof-of-use

### PR DESCRIPTION
**Pull request by ritave**

### CODE REVIEW

File: packages/signing-server/src/server.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    demandOption: true,
+  })
+  .option("nodeEndpoint", {
+    describe: "Ethereum node url",
+    requiresArg: true,
+    demandOption: true,
+  })
+  .strict().argv;
+
+const port = Number.parseInt(args.port, 10);
```
Comment (dankins) [inline packages/signing-server/src/server.ts]: this should probably be an environment variable, I think NODE_PORT is the most common

File: packages/signing-server/src/server.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    await civil.accountStream.first().toPromise();
+    const userGroups = await civil.userGroupsSingletonTrusted();
+    const nonce = await userGroups.getMaxGroupSizeUpdateNonce();
+    const signature = actionSigner.signMaxGroupSize(toBuffer(userGroups.address), new BN(nonce), new BN(size));
+    const twoStep: TwoStepEthTransaction = await userGroups.setMaxGroupSize(Number.parseInt(size, 10), signature);
+    ctx.body = twoStep.txHash;
+  }),
+);
+
+app.listen(port);
```
Comment (dankins) [inline packages/signing-server/src/server.ts]: this needs to bind to 0.0.0.0, and should be configurable by the NODE_HOST env var

File: packages/signing-server/src/server.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    if (!isValidAddress(groupA) || !isValidAddress(groupB)) {
+      throw new Error("Invalid address");
+    }
+    // Due to interactions with core, web3 doesn't have a default account when trying to send tx
+    // Ensure it's set for the first time
+    await civil.accountStream.first().toPromise();
+    const userGroups = await civil.userGroupsSingletonTrusted();
+    const signature = actionSigner.signUnion(toBuffer(userGroups.address), toBuffer(groupA), toBuffer(groupB));
+    const twoStep: TwoStepEthTransaction = await userGroups.forceUnion(groupA, groupB, signature);
+    ctx.body = twoStep.txHash;
```
Comment (dankins) [inline packages/signing-server/src/server.ts]: can we wrap this in an object so it is returned as JSON instead of just a string?

Comment (ritave) [reply packages/signing-server/src/server.ts]: what structure would it be? `{ hash: "hash here" }`?

File: packages/signing-server/src/server.ts (hunk 1 above)
Comment (ritave) [reply packages/signing-server/src/server.ts]: Yargs enables env parsing.
`SIGNING_SERVER_PORT=1234` will work, similar for all other params

File: packages/signing-server/src/server.ts (hunk 3 above)
Comment (dankins) [reply packages/signing-server/src/server.ts]: yea, maybe { "txHash": "hash here"} to make it more explicit as to what the hash is

**Pull request merged**